### ISSUE TITLE
Record accessors with hyphens/dashes not working in grep filters

### ISSUE THREAD

**Issue by nemanjamart** [aws/aws-for-fluent-bit#311]
### Describe the question/issue

Record accessors in filters with hyphens/dashes causing fluent-bit pods to error out.

### Configuration

Create a grep filter with a record accessor that contains hyphens/dashes (like below: X-Amzn-Trace-Id):

    [FILTER]
        Name           grep
        Match          kube.*
        Exclude        X-Amzn-Trace-Id readinessProbe|livenessProbe

### Fluent Bit Log Output 

Run fluent-bit and it will fail with the following message:

[error] [sp] bad input character '-' at line 0
[error] [record accessor] syntax error, unexpected IDENTIFIER, expecting $end at '$X-Amzn-Trace-Id'
[error] [filter:grep:grep.5] invalid record accessor? '$X-Amzn-Trace-Id'
[error] Failed initialize filter grep.5

### Fluent Bit Version Info

Exists in 2.10.1 and 2.22.0, no error in 2.1.1

### Related Issues

Already reported here:
https://github.com/fluent/fluent-bit/issues/3063

**Comment by zhonghui12** [comment]
Hi @nemanjamart , I wonder if the key should be specified as `$X-Amzn-Trace-Id`?

**Comment by nemanjamart** [comment]
Hi @zhonghui12 , thanks for the reply. Unfortunately, it is not working:
```
    [FILTER]
        Name           grep
        Match          kube.*
        Exclude        $X-Amzn-Trace-Id readinessProbe|livenessProbe
```
```
[2022/03/11 00:31:17] [ info] [filter:kubernetes:kubernetes.0] local POD info OK
[2022/03/11 00:31:17] [ info] [filter:kubernetes:kubernetes.0] testing connectivity with API server...
[2022/03/11 00:31:17] [error] [lib] backend failed
[2022/03/11 00:31:17] [ info] [filter:kubernetes:kubernetes.0] connectivity OK
[2022/03/11 00:31:17] [error] [record accessor] bad input character '-' at line 0
[2022/03/11 00:31:17] [engine] caught signal (SIGSEGV)
AWS for Fluent Bit Container Image Version 2.22.0
```

I am running it successfully with version 2.1.1 (same config, just without $ as in original post). But I think this should be fixed in newer versions (like 2.22.0)

**Comment by nemanjamart** [comment]
Just tried other variations - no luck:
For example:
```
    [FILTER]
        Name           grep
        Match          kube.*
        Exclude        $['X-Amzn-Trace-Id'] readinessProbe|livenessProbe
```


```
[2022/03/11 00:34:33] [error] [record accessor] syntax error, unexpected '[', expecting IDENTIFIER at '$['X-Amzn-Trace-Id']'
[2022/03/11 00:34:33] [error] [filter:grep:grep.5] invalid record accessor? '$['X-Amzn-Trace-Id']'
[2022/03/11 00:34:33] [error] Failed initialize filter grep.5
[2022/03/11 00:34:33] [ info] [input] pausing tail.0
[2022/03/11 00:34:33] [error] [lib] backend failed
[2022/03/11 00:34:33] [ info] [input] pausing tail.1
[2022/03/11 00:34:33] [engine] caught signal (SIGSEGV)
AWS for Fluent Bit Container Image Version 2.22.0
```

**Comment by nemanjamart** [comment]
Aha ok, I was digging a bit around - the PR with the fix is not yet merged into the codebase:

https://github.com/fluent/fluent-bit/pull/4263

Hoping that this solves accessors hyphen/dash issues as well!

**Comment by nemanjamart** [comment]
Ok, I see that this is part of fluent-bit 1.9.0-rc4 now.

**Comment by zhonghui12** [comment]
@nemanjamart  I see. So I missed the point that this fix was only merged into main branch which means it should be included in next major release Fluent Bit 1.9 and current is 1.8. FLB 1.9 should come soon and we will cut a release based on it after it is out. Thanks for your patience and I will let you know the fix is out.

**Comment by zhonghui12** [comment]
This issue should be fixed as we already release based on FLB 1.9. Please let me know if there are more problems on this issue.

**Comment by ZenSecurity** [comment]
@zhonghui12 what is about symbol `@` ?
I catched the following problem:
`[error] [record accessor] bad input character '@' at line 0`
